core/src/test/scala/org/apache/spark/sql/delta/DeltaSuite.scala

```diff
@@ -1643,6 +1643,38 @@ class DeltaSuite extends QueryTest
     dir2.renameTo(dir1)
     checkAnswer(spark.read.format("delta").load(dir1.getCanonicalPath), spark.range(10).toDF)
   }
+
+  test("set metadata upon write") {
+    withTempDir { tempDir =>
+      val path = tempDir.getCanonicalPath
+      val options = Map(
+        "header"-> "true",
```

Comment:
After syncing with @zsxwing , we've confirmed that we do not want to write properties that are not prefixed with `delta`. Thus, to be clear, we only want to write to table metadata those properties which start with `delta`.